docs/user-guide/materialization.md

```diff
@@ -0,0 +1,236 @@
+# Materialization
+
+## What is materialization?
+
+Capturing and storing the nix files for a project so that they do
+not need to be built (or checked).
```

Comment:
Also known as caching IFDs.

Maybe we should note that explicitly for someone who looks for caching or IFDs?